### PR TITLE
Fixes 1471: Integrate snapshot task into application

### PR DESCRIPTION
**Pull request by jlsherrill**
## Summary

This adds snapshotting to repositories, integrated with the api and tasking system.  When creating a repository, if you pass snapshot=true, a task is spawned to snapshot the repository.  This snapshot task should save information within the payload such as sync task href, publication task href, etc..., such that if the worker dies, the task should be able to be resumed.

## Testing steps

Run a create or bulk_create request as normal, but add the parameter 'snapshot=true'.

Try killing the worker (with ctrl-c or kill -9) and restarting it.  In all cases you should be able to see the task resumed and a snapshot created.  For now you'll have to check the db to see if snapshot got created:

```select distribution_path from snapshots;```

you can then curl the pulp distribution path:

```curl  localhost:8080/pulp/content/73e2d9fa-387d-4df1-ac58-fe898aebb30f/f4e2f625-4cba-4d5c-9ea5-3de353bb72f1/```

For QE:

We'll need to update automation to support the snapshot parameter, but we can't really test snapshots yet, as there is no way to monitor the snapshot or confirm its created.

### CODE REVIEW

**Comment by jlsherrill** [comment]
https://issues.redhat.com/browse/HMS-1471

File: deployments/deployment.yaml [hunk 1]
```diff
@@ -296,3 +336,13 @@ parameters:
   - name: FLOORIST_QUERY_PREFIX
     description: Prefix for separating query data between prod and stage in the bucket
     value: "hms_analytics/content-sources/unknown"
+  - name: CLIENTS_PULP_SERVER
+    description: Pulp Api URL (i.e. http://hostname:8080)
```
Comment (swadeley) [inline deployments/deployment.yaml:340]:
```suggestion
    description: Pulp Api URL (e.g. http://hostname:8080)
```

**Review by Andrewgdewar — DISMISSED** [review]
Ack for functionality.

**Review by rverdile — COMMENTED** [review]
those are my only two comments!

**Comment by swadeley** [comment]
Hi, I have tested this and updated automation:

[Add snapshot bool (!180) · Merge requests · insights-qe / iqe-content-sources-plugin · GitLab](https://gitlab.cee.redhat.com/insights-qe/iqe-content-sources-plugin/-/merge_requests/180)

File: deployments/deployment.yaml [hunk 2]
```diff
@@ -275,7 +315,7 @@ parameters:
   - name: MEMORY_REQUESTS
     value: 100Mi
   - name: LOGGING_LEVEL
-    value: warn
+    value: debug
```
[the remark this answers is not in the document]
Comment (jlsherrill) [reply deployments/deployment.yaml:318]: ah yes, it wasn't, thx!

File: pkg/tasks/repository_snapshot.go [hunk 3]
```diff
@@ -1,50 +1,84 @@
...
-// SnapshotOptions Options for the snapshot task.
-//
-//	These may change with integration with the tasking system
-type SnapshotOptions struct {
-	OrgId          string
-	RepoConfigUuid string
-	DaoRegistry    *dao.DaoRegistry
-	PulpClient     pulp_client.PulpClient
-	snapshotIdent  *string // Optional, autogenerated
+const Snapshot = "Snapshot"
```
[the remark this answers is not in the document]
Comment (jlsherrill) [reply pkg/tasks/repository_snapshot.go]: good catch, changed to lowercase

**Comment by jlsherrill** [comment]
i'll want to disable the pulpserver in stage and prod before we merge this, openeing an MR to do that.

**Review by rverdile — APPROVED** [review]
lgtm!

**Pull request merged**